src/data/index.ts

```diff
@@ -4,8 +4,10 @@ import * as metric from "@src/data/metric";
 import * as ping from "@src/data/ping";
 import * as tag from "@src/data/tag";
 import * as target from "@src/data/target";
+import * as upgrade from "@src/data/upgrade";
 import * as upload from "@src/data/upload";
 export { tag } from "@src/data/tag";
+export { upgrade } from "@src/data/upgrade";
 
 export function start(): void {
```

Comment:
What determines which of reset/start/end is used in thestart() and end() functions below? Wasn't quite sure of the logic.

Reply:
I can walk you through it offline. Anything that just resets the state, the module exposes "reset" function. Anything that needs to initialize some variables and then later clear state, those modules use start / end. Should we later unify for consistency? Probably. 

---
In reply to: [368679796](https://github.com/microsoft/clarity-js/pull/166#discussion_r368679796) [](ancestors = 368679796)